src/features/profile/ProfileSetup.tsx

```
@@ -0,0 +1,467 @@
+		(field) => form.state.values[field]?.length > 0,
+	).length;
+	const completedOptional = optionalFields.filter((field) => {
+		const value = form.state.values[field];
+		return typeof value === "string" && value.length > 0;
+	}).length;
+	const completionPercentage =
+		((completedRequired * 2 + completedOptional) /
+			(requiredFields.length * 2 + optionalFields.length)) *
+		100;
+
+	const handleAvatarUpload = useCallback(
```

Biome (1.9.4)

[error] This hook specifies more dependencies than necessary: userId

This dependency can be removed from the list.


(lint/correctness/useExhaustiveDependencies)